File: update_migrations.py
import os
import glob
import re

migrations_dir = 'v2/database/migrations/'
files = glob.glob(migrations_dir + '*.php')

schemas = {
    'create_users_table': '''
            \->id();
            \->string('name');
            \->string('email')->unique()->nullable();
            \->string('username')->unique();
            \->timestamp('email_verified_at')->nullable();
            \->string('password');
            \->enum('role', ['admin', 'teacher', 'student'])->default('student');
            \->rememberToken();
            \->timestamps();
''',
    'create_students_table': '''
            \->id();
            \->foreignId('user_id')->constrained('users')->cascadeOnDelete();
            \->string('nis', 25)->unique();
            \->string('nisn', 20)->nullable();
            \->string('name', 150);
            \->string('phone', 20)->nullable();
            \->text('address')->nullable();
            \->foreignId('school_class_id')->nullable()->constrained('school_classes')->nullOnDelete();
            \->string('gender', 10)->nullable();
            \->string('religion', 50)->default('Islam');
            \->timestamps();
''',
    'create_teachers_table': '''
            \->id();
            \->foreignId('user_id')->constrained('users')->cascadeOnDelete();
            \->string('nip', 50)->unique()->nullable();
            \->string('name', 150);
            \->string('phone', 20)->nullable();
            \->text('address')->nullable();
            \->string('employment_status', 20)->nullable();
            \->boolean('is_bk')->default(false);
            \->timestamps();
''',
    'create_school_classes_table': '''
            \->id();
            \->string('name', 100);
            \->foreignId('teacher_id')->nullable()->constrained('teachers')->nullOnDelete();
            \->timestamps();
''',
    'create_subjects_table': '''
            \->id();
            \->string('name', 100);
            \->timestamps();
''',
    'create_schedules_table': '''
            \->id();
            \->foreignId('subject_id')->constrained('subjects')->cascadeOnDelete();
            \->foreignId('teacher_id')->constrained('teachers')->cascadeOnDelete();
            \->foreignId('school_class_id')->constrained('school_classes')->cascadeOnDelete();
            \->string('day', 20);
            \->time('start_time');
            \->time('end_time');
            \->string('room', 80)->nullable();
            \->timestamps();
''',
    'create_attendances_table': '''
            \->id();
            \->foreignId('schedule_id')->nullable()->constrained('schedules')->nullOnDelete();
            \->foreignId('student_id')->constrained('students')->cascadeOnDelete();
            \->date('date');
            \->string('status', 20)->default('present'); // present, absent, sick, permission
            \->text('notes')->nullable();
            \->timestamps();
''',
    'create_journals_table': '''
            \->id();
            \->foreignId('schedule_id')->nullable()->constrained('schedules')->nullOnDelete();
            \->foreignId('teacher_id')->constrained('teachers')->cascadeOnDelete();
            \->date('date');
            \->text('material_covered')->nullable();
            \->text('notes')->nullable();
            \->timestamps();
'''
}

for filepath in files:
    filename = os.path.basename(filepath)
    content = open(filepath, 'r').read()
    
    for key, schema_code in schemas.items():
        if key in filename:
            # Replace empty Schema::create block with new schema
            pattern = r"(Schema::create\('.*?', function \(Blueprint \\) \{).*?(\}\);)"
            replacement = r"\g<1>\n" + schema_code + r"        \g<2>"
            new_content = re.sub(pattern, replacement, content, flags=re.DOTALL)
            open(filepath, 'w').write(new_content)
            print("Updated " + filename)

File: update_migrations.php
<?php
$migrationsDir = __DIR__ . '/v2/database/migrations/';
$files = glob($migrationsDir . '*.php');

$schemas = [
    'create_users_table' => <<<PHP
            \$table->id();
            \$table->string('name');
            \$table->string('email')->unique()->nullable();
            \$table->string('username')->unique();
            \$table->timestamp('email_verified_at')->nullable();
            \$table->string('password');
            \$table->enum('role', ['admin', 'teacher', 'student'])->default('student');
            \$table->rememberToken();
            \$table->timestamps();
PHP,
    'create_students_table' => <<<PHP
            \$table->id();
            \$table->foreignId('user_id')->constrained('users')->cascadeOnDelete();
            \$table->string('nis', 25)->unique();
            \$table->string('nisn', 20)->nullable();
            \$table->string('name', 150);
            \$table->string('phone', 20)->nullable();
            \$table->text('address')->nullable();
            \$table->foreignId('school_class_id')->nullable()->constrained('school_classes')->nullOnDelete();
            \$table->string('gender', 10)->nullable();
            \$table->string('religion', 50)->default('Islam');
            \$table->timestamps();
PHP,
    'create_teachers_table' => <<<PHP
            \$table->id();
            \$table->foreignId('user_id')->constrained('users')->cascadeOnDelete();
            \$table->string('nip', 50)->unique()->nullable();
            \$table->string('name', 150);
            \$table->string('phone', 20)->nullable();
            \$table->text('address')->nullable();
            \$table->string('employment_status', 20)->nullable();
            \$table->boolean('is_bk')->default(false);
            \$table->timestamps();
PHP,
    'create_school_classes_table' => <<<PHP
            \$table->id();
            \$table->string('name', 100);
            \$table->foreignId('teacher_id')->nullable()->constrained('teachers')->nullOnDelete();
            \$table->timestamps();
PHP,
    'create_subjects_table' => <<<PHP
            \$table->id();
            \$table->string('name', 100);
            \$table->timestamps();
PHP,
    'create_schedules_table' => <<<PHP
            \$table->id();
            \$table->foreignId('subject_id')->constrained('subjects')->cascadeOnDelete();
            \$table->foreignId('teacher_id')->constrained('teachers')->cascadeOnDelete();
            \$table->foreignId('school_class_id')->constrained('school_classes')->cascadeOnDelete();
            \$table->string('day', 20);
            \$table->time('start_time');
            \$table->time('end_time');
            \$table->string('room', 80)->nullable();
            \$table->timestamps();
PHP,
    'create_attendances_table' => <<<PHP
            \$table->id();
            \$table->foreignId('schedule_id')->nullable()->constrained('schedules')->nullOnDelete();
            \$table->foreignId('student_id')->constrained('students')->cascadeOnDelete();
            \$table->date('date');
            \$table->string('status', 20)->default('present');
            \$table->text('notes')->nullable();
            \$table->timestamps();
PHP,
    'create_journals_table' => <<<PHP
            \$table->id();
            \$table->foreignId('schedule_id')->nullable()->constrained('schedules')->nullOnDelete();
            \$table->foreignId('teacher_id')->constrained('teachers')->cascadeOnDelete();
            \$table->date('date');
            \$table->text('material_covered')->nullable();
            \$table->text('notes')->nullable();
            \$table->timestamps();
PHP,
];

foreach ($files as $file) {
    $filename = basename($file);
    $content = file_get_contents($file);
    
    foreach ($schemas as $key => $schema) {
        if (strpos($filename, $key) !== false) {
            // Find Schema::create block
            $pattern = '/(Schema::create\([\s\S]*?function\s*\(\s*Blueprint\s*\$table\s*\)\s*\{)([\s\S]*?)(\}\);)/';
            $replacement = "$1\n" . $schema . "\n        $3";
            $newContent = preg_replace($pattern, $replacement, $content);
            file_put_contents($file, $newContent);
            echo "Updated $filename\n";
        }
    }
}
